### PR TITLE
[WIP] add auto publishing

### LINKED ISSUE #37: Publish Stable and Snapshots to jfrog via Travis
in the lookout org.

SNAPSHOT to jfrog
releases to bintray (maybe via jfrog)

### PR DESCRIPTION
**Pull request by jamescway**
*Work in progress*.  Here is a first pass for auto publishing artifacts to bintray.  It does not address snapshots.  It uses the sbt-release plugin to help do the heavy lifting.
* runs the tests
* runs publish (to bintray) strips "SNAPSHOT" from version
* increments bugfix version in version.sbt and adds SNAPSHOT back
* pushes updated code back to github repo
* only runs if you tag it by doing `git tag <some-tag>` then doing `git push origin --tags`

Fixes #37

### CODE REVIEW

**Comment by maheshkelkar** [comment]
@jamescway I just wanted to understand impact of the following operations on the version numbers or these new files. Or the things that user should be aware of as a result of the following:
1. normal commits to pull request, (a) I don't want to bump up version, (b) I want to bump up the version
2. merging the pull request
3. want to release a SNAPSHOT version, what commands do I type?
4. I want to release a version to BINTRAY, what commands should I type?

File: .travis.yml [hunk 1]
```diff
@@ -18,3 +29,8 @@ notifications:
     on_success: change  # options: [always|never|change] default: always
     on_failure: always  # options: [always|never|change] default: always
     on_start: false     # default: false
+before_script:
+  - git remote set-url origin "https://${GH_TOKEN}@github.com/${TRAVIS_REPO_SLUG}.git"
+before_install:
+  - git config --global user.email "${GH_USER_EMAIL}"
+  - git config --global user.name "${GH_USER}"
```
Comment (maheshkelkar) [inline .travis.yml]: When does user execute this shell script? Only once while setting up the repository? Or everytime you want to release the version? etc?

File: build.sbt [hunk 2]
```diff
@@ -66,7 +65,7 @@ lazy val publishSettings = Seq(
   ),
   bintrayRepository := "borderpatrol",
   publishMavenStyle := true,
-  bintrayOrganization := Some("lookout")
+  bintrayOrganization := Some("jamescway")
```
Comment (maheshkelkar) [inline build.sbt]: I assume you are doing this because you are testing it in your account, right?

File: version.sbt [hunk 3]
```diff
@@ -0,0 +1 @@
+version in ThisBuild := "0.1.3-SNAPSHOT"
```
Comment (maheshkelkar) [inline version.sbt]: Does user modify this file or is it automatically updated?

File: .travis.yml [hunk 4]
```diff
@@ -8,9 +8,20 @@ jdk:
...
+    secure: "Z7kKcMOFR4Erf5p3uh8XN4zePTg+rEjQmujhleosKbCLNzkEA/aupP5oMk/NXcpBJtVlNDqWcM26uhvs9Ddp5bJTY/BgldJvmolS+2B9jJqfsOtV//Dpp9Ty8h2kWh4OigOxJc5OvyWvgdW6ntHQ1cGM/t0K1Hd71niUN1sHDnXtejKIr6oZcP1UkObqe9d1TffYNaXwD2a0rguIGbYbdPdmMPEygmhpZvFg0VE8Xq68666wzgTi1ExDwq0e2PhcDq69qgx5YfqthXj8YuytzTQto2lGWugAqhJEhlPKLwYS1TcowTyYXl9WkC9ouoxYQeQ1qvuA3S7lJBbyEyEsEAFt4fZMqLxKpCMq5LtnlTUk3vQW8XYG4Cg15JN4E7oPHbIqP+Bq9eVCu1vGI2/NjhIpUyRojNI0tQ+wTcC9Z7ilL31a98KJTYuZf3JAnCcqx1FjTfw69UaFqGgIsVOsnv14FNfLnn2DBtJgYJTvrl1PCigwi0se6RZHEuk13VAD+uolV8eWz442QK8l6wtnOpy57GAUX9Esw5CBlpM6Psg+MoWy8JeH127dsHb6ah6YKnCa/807CbXQ+wKI5WtBLeRZlgxxooQcSjAeNPZF5hnJLtF2OgIYzvY8sNwrs489mTyp84pziQraZr0nPGbcIrAQUHadb+RbBUnQRu2E/lQ="
```
Comment (maheshkelkar) [inline .travis.yml]: Did you add this by executing "travis cli command" to encrypt bintray & sonatype account usernames & passwords?

File: .travis.yml (hunk 1 above)
Comment (jamescway) [inline .travis.yml]: this is to setup the github credentials for when the release plugin commits to the repo to increment the version.sbt to the next version.

File: build.sbt (hunk 2 above)
Comment (jamescway) [inline build.sbt]: yep

**Comment by jamescway** [comment]
1. normal commits to pull request, (a) I don't want to bump up version, (b) I want to bump up the version
>>normal commits won't bump version, because you didn't tag it, it should auto bump the bugfix version (major.minor.bugfix) this can be modified in settings
2. merging the pull request
>>only a snapshot is released for every merge to master, no official release
3. want to release a SNAPSHOT version, what commands do I type?
>>with this processes it should be automatic with every merge to master
4. I want to release a version to BINTRAY, what commands should I type?
>>you push a tag.  `git tag v1.1.1` `git push origin --tags`

**Comment by maheshkelkar** [comment]
@jamescway One more question: I see that you have moved "version" from buildSettings in build.sbt to "version.sbt file. Is it common practice? How does SBT determine that?

**Comment by jamescway** [comment]
@maheshkelkar The sbt-release plugin uses the version.sbt as part of its processes.  Its allows easier modifying when auto-incrementing version number.  Its analogous to ruby's version.rb.  I've seen some release plugins use version.txt also.

**Comment by maheshkelkar** [comment]
We are shelving this as support for bintray & jfrog was added using the pull request #98

**Pull request closed without merging**